rancher2/schema_node_template.go

```diff
@@ -12,15 +12,37 @@ type NodeTemplate struct {
 	Amazonec2Config     *amazonec2Config     `json:"amazonec2Config,omitempty" yaml:"amazonec2Config,omitempty"`
 	AzureConfig         *azureConfig         `json:"azureConfig,omitempty" yaml:"azureConfig,omitempty"`
 	DigitaloceanConfig  *digitaloceanConfig  `json:"digitaloceanConfig,omitempty" yaml:"digitaloceanConfig,omitempty"`
+	HarvesterConfig     *harvesterConfig     `json:"harvesterConfig,omitempty" yaml:"harvesterConfig,omitempty"`
+	HetznerConfig       *hetznerConfig       `json:"hetznerConfig,omitempty" yaml:"hetznerConfig,omitempty"`
 	LinodeConfig        *linodeConfig        `json:"linodeConfig,omitempty" yaml:"linodeConfig,omitempty"`
+	OpennebulaConfig    *opennebulaConfig    `json:"opennebulaConfig,omitempty" yaml:"opennebulaConfig,omitempty"`
 	OpenstackConfig     *openstackConfig     `json:"openstackConfig,omitempty" yaml:"openstackConfig,omitempty"`
 	VmwarevsphereConfig *vmwarevsphereConfig `json:"vmwarevsphereConfig,omitempty" yaml:"vmwarevsphereConfig,omitempty"`
-	OpennebulaConfig    *opennebulaConfig    `json:"opennebulaConfig,omitempty" yaml:"opennebulaConfig,omitempty"`
-	HetznerConfig       *hetznerConfig       `json:"hetznerConfig,omitempty" yaml:"hetznerConfig,omitempty"`
 }
 
 //Schemas
 
+var allNodeTemplateDriverConfigFields = []string{
+	"amazonec2_config",
+	"azure_config",
+	"digitalocean_config",
+	"harvester_config",
+	"hetzner_config",
+	"linode_config",
+	"opennebula_config",
+	"openstack_config",
+	"vsphere_config"}
+
+func getConflicts(fieldNames []string, fieldName string) []string {
```

Comment:
This function should be moved to the first commit so as to not break things like `git bicect` and `git revert`.

Reply:
@thedadams Updated